### PR TITLE
Update to support new lib-common storage structs

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: apis/go.mod [hunk 1]
```diff
@@ -115,3 +115,5 @@ replace github.com/openshift/api => github.com/openshift/api v0.0.0-202408300231
 
 // custom RabbitmqClusterSpecCore for OpenStackControlplane (v2.9.0_patches_tag)
 replace github.com/rabbitmq/cluster-operator/v2 => github.com/openstack-k8s-operators/rabbitmq-cluster-operator/v2 v2.6.1-0.20241017142550-a3524acedd49 //allow-merging
+
+replace github.com/openstack-k8s-operators/lib-common/modules/storage => github.com/dprince/lib-common/modules/storage v0.0.0-20241026143649-333dbab71ed9
```
Comment (stuggi) [inline apis/go.mod:119]: its merged now

**Comment by dprince** [comment]
One more fix is needed here for this to pass: https://github.com/openstack-k8s-operators/lib-common/pull/576

**Review by jpodivin — APPROVED** [review]

**Comment by dprince** [comment]
kuttl test failures on ci/prow/openstack-operator-build-deploy-kuttl  look valid

**Comment by dprince** [comment]
recheck

**Comment by dprince** [comment]
recheck

**Comment by bshephar** [comment]
Seems to be the same error here:
```
❯ curl -s "https://storage.googleapis.com/test-platform-results/pr-logs/pull/openstack-k8s-operators_openstack-operator/1158/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1852159512341909504/build-log.txt" | grep "missing from
 map"
    case.go:380: resource OpenStackDataPlaneNodeSet:openstack-kuttl-tests/edpm-extramounts: .spec.nodeTemplate.extraMounts.volumes.persistentVolumeClaim: key is missing from map
    case.go:380: resource Job:openstack-kuttl-tests/test-service-edpm-extramounts-edpm-extramounts: .spec.template.spec.volumes.persistentVolumeClaim: key is missing from map
    case.go:380: resource Job:openstack-kuttl-tests/test-service-edpm-extramounts-edpm-extramounts: .spec.template.spec.volumes.persistentVolumeClaim: key is missing from map
```

**Comment by dprince** [comment]
recheck

**Comment by jpodivin** [comment]
There seems to be genuine failure in the glance adoption:
```
```

**Comment by jpodivin** [comment]
There appears to be a genuine failure in the glance adoption process: 
```
{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "GlanceAPI": {"name":"glance-default-internal","namespace":"openstack"}, "namespace": "openstack", "name": "glance-default-internal", "reconcileID": "26a4908c-b5c4-40ff-8e4e-bddcbe4c3e87", "error": "not all pods have interfaces with ips as configured in NetworkAttachments: [storage]"}
```

**Comment by fmount** [comment]
note: my last kuttl check might result in the same failure of the previous run [1], because `go.mod` should be refreshed to reflect the last hash associated to the lib-common update.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_openstack-operator/1158/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1852671147426975744/artifacts/openstack-operator-build-deploy-kuttl/openstack-k8s-operators-kuttl/build-log.txt

**Comment by dprince** [comment]
recheck

**Comment by fmount** [comment]
@dprince from [1] I do not see an explicit kuttl failure like before. I assume we need to simply recheck.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_openstack-operator/1158/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1853449635142569984/artifacts/openstack-operator-build-deploy-kuttl/openstack-k8s-operators-kuttl/build-log.txt

**Review by fmount — APPROVED** [review]
/lgtm

**Pull request merged**